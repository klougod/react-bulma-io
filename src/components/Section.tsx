import React from 'react'
import PropTypes from 'prop-types'

export const Section = ({ className, children }: any) => (
  <section className={`section ${className}`}>{children}</section>
)

Section.defaultProps = {
  className: ''
}

Section.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired
}

export default Section
