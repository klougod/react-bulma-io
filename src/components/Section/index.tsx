import React, { ReactNode } from 'react'

interface SectionProps {
  className?: string
  children?: ReactNode
}
export const Section = ({ className, children }: SectionProps) => (
  <section className={`section ${className}`}>{children}</section>
)

Section.defaultProps = {
  className: ''
}

export default Section
