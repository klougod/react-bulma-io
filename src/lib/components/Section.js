import React from 'react'
import PropTypes from 'prop-types'

export const Section = props => {

  const {
    className,
    children
  } = props

  return (
    <section className={`section ${className}`}>
      { children }
    </section>
  )
}

Section.defaultProps = {
  className: ''
}

Section.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired
}

export default Section