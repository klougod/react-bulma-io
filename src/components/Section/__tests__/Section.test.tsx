import React from 'react'
import Enzyme from 'enzyme'

import Section from '../'

describe('Section component test', function () {
  it('should render without crash without props', function () {
    const sectionMock = Enzyme.shallow(<Section />)
    expect(sectionMock).toMatchSnapshot()
  })

  it('should render without crash with props', function () {
    const sectionMockProps = {
      className: 'sectionCLass'
    }
    const sectionMock = Enzyme.shallow(<Section {...sectionMockProps} />)
    expect(sectionMock).toMatchSnapshot()
  })
})
