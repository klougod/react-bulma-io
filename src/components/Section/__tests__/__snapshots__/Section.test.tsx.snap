// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section component test should render without crash with props 1`] = `
<section
  className="section sectionCLass"
/>
`;

exports[`Section component test should render without crash without props 1`] = `
<section
  className="section "
/>
`;
